### PR TITLE
correct error introduced in add_torrent_impl refactor, with regards t…

### PR DESCRIPTION
**Pull request by arvidn**
…o not having flag_duplicate_is_error set (which is not set by default)

### CODE REVIEW

File: src/session_impl.cpp [hunk 1]
```diff
@@ -5086,19 +5093,19 @@ namespace aux {
 					torrent_ptr->set_url(params.url);
 				if (!params.source_feed_url.empty() && torrent_ptr->source_feed_url().empty())
 					torrent_ptr->set_source_feed_url(params.source_feed_url);
-				return torrent_ptr;
+				return std::make_pair(torrent_ptr, false);
```
Comment (gubatron) [inline src/session_impl.cpp]: does this mean...
`if ((params.flags & add_torrent_params::flag_duplicate_is_error) == 0)`

that duplicates aren't errors?

If so, is it ok to put `false` in the second element of the pair?

Comment (gubatron) [inline src/session_impl.cpp]: (I say this because I notice you put `false` whenever there's an error)

File: src/session_impl.cpp [hunk 2]
```diff
@@ -5086,19 +5093,19 @@ namespace aux {
...
 			}
 
 			ec = errors::duplicate_torrent;
-			return ptr_t();
+			return std::make_pair(ptr_t(), false);
```
Comment (gubatron) [inline src/session_impl.cpp]: here it looks good to me.

File: src/session_impl.cpp (hunk 1 above)
Comment (gubatron) [inline src/session_impl.cpp]: hmm, I see the boolean probably means `started`

**Comment by aldenml** [comment]
It's not clear to me why the line `if (!added) return handle;` is placed where it is. It is after the "add_peers" logic, but before the "add_dht_nodes" logic. I don't see any significant issue here, just to left the observation.

File: src/session_impl.cpp (hunk 1 above)
Comment (arvidn) [inline src/session_impl.cpp]: yeah, it's mimicking std::set<>::insert()

**Comment by arvidn** [comment]
yeah, adding the DHT nodes should probably be done earlier. It's important to exit here though before calling start() and adding extensions, things that actually violate the invariants.

**Comment by aldenml** [comment]
@arvidn try to clean the cache for https://travis-ci.org/arvidn/libtorrent/jobs/139678216

**Pull request merged**